### ISSUE TITLE
Problems with decompiling new Vindictus maps

### ISSUE THREAD

**Issue by MARK2580** [ata4/bspsrc#136]
When decompiling new Vindictus maps, a lot of errors appear, they seem to be related to changes in the statics of the models in the game. Such vmf are not finished yet.
Archive with examples bsp - https://disk.yandex.ru/d/Kxb3-aRqcGzalQ

**Here's an example of errors.**
```
[warning] VmfWriter: Unclosed VMF chunk: entity
[error] BspSource: Decompiling error, caused by java.lang.IndexOutOfBoundsException: Index: 52429, Size: 45
	at java.util.ArrayList.rangeCheck(Unknown Source)
	at java.util.ArrayList.get(Unknown Source)
	at info.ata4.bspsrc.modules.entity.EntitySource.writeStaticProps(EntitySource.java:686)
	at info.ata4.bspsrc.modules.BspDecompiler.writeEntities(BspDecompiler.java:176)
	at info.ata4.bspsrc.modules.BspDecompiler.start(BspDecompiler.java:102)
	at info.ata4.bspsrc.BspSource.decompile(BspSource.java:164)
	at info.ata4.bspsrc.BspSource.run(BspSource.java:72)
	at info.ata4.bspsrc.gui.BspSourceFrame$4.run(BspSourceFrame.java:359)
```

```
[warning] BspFileReader: Couldn't find static prop struct for version 7
[warning] BspFileReader: Falling back to static prop v4
[error] BspFileReader: Lump reading error in sprp, caused by java.io.IOException: 4 bytes remaining
	at info.ata4.bsplib.BspFileReader.checkRemaining(BspFileReader.java:948)
	at info.ata4.bsplib.BspFileReader.loadStaticProps(BspFileReader.java:461)
	at info.ata4.bsplib.BspFileReader.loadAll(BspFileReader.java:88)
	at info.ata4.bspsrc.BspSource.decompile(BspSource.java:123)
	at info.ata4.bspsrc.BspSource.run(BspSource.java:72)
	at info.ata4.bspsrc.gui.BspSourceFrame$4.run(BspSourceFrame.java:359)
```

```
[warning] VmfWriter: Unclosed VMF chunk: entity
[error] BspSource: Decompiling error, caused by java.lang.IndexOutOfBoundsException: Index: 13107, Size: 10
	at java.util.ArrayList.rangeCheck(Unknown Source)
	at java.util.ArrayList.get(Unknown Source)
	at info.ata4.bspsrc.modules.entity.EntitySource.writeStaticProps(EntitySource.java:686)
	at info.ata4.bspsrc.modules.BspDecompiler.writeEntities(BspDecompiler.java:176)
	at info.ata4.bspsrc.modules.BspDecompiler.start(BspDecompiler.java:102)
	at info.ata4.bspsrc.BspSource.decompile(BspSource.java:164)
	at info.ata4.bspsrc.BspSource.run(BspSource.java:72)
	at info.ata4.bspsrc.gui.BspSourceFrame$4.run(BspSourceFrame.java:359)
```

**Comment by rihi** [comment]
I didn't test it with the given bsp's, but I believe this has been fixed with #92. The fix is not in the 1.4.0 version, so it's not yet available.

**Comment by MARK2580** [comment]
> I didn't test it with the given bsp's, but I believe this has been fixed with #92. The fix is not in the 1.4.0 version, so it's not yet available.

I think because your post was written a whole month after the release of version 1.4.0

**Comment by rihi** [comment]
Hey, the code from #92 has now been released with version 1.4.1.
If your problem is not resolve with that new version, feel free to write here and I'll have another look at the problem.
